src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -102,7 +102,17 @@ public bool StartAssignment(HostAssignmentContext context)
         {
             if (!_webHostEnvironment.InStandbyMode)
             {
-                _logger.LogError("Assign called while host is not in placeholder mode");
+                // This is only true when specializing pinned containers.
+                if (!context.Environment.TryGetValue(EnvironmentSettingNames.ContainerStartContext, out string startContext))
```

Comment:
>if (!context.Environment.TryGetValue(EnvironmentSettingNames.ContainerStartContext, out string startContext)) [](start = 16, length = 109)

Do we need to handle scenario where Env setting is present but startContext can still be null or empty? (return false in that case, right?)

Reply:
If start context is null or empty, that means it's a placeholder pool container.